aggregator/src/lib.rs

```diff
@@ -0,0 +1,29 @@
+use std::{fmt::Debug, hash::Hash};
+
+use aleph_bft::Recipient;
+use aleph_bft_rmc::Message;
+
+mod aggregator;
+mod multicast;
+
+pub use crate::{
+    aggregator::{BlockSignatureAggregator, IO},
+    multicast::SignableHash,
+};
+
+pub type RmcNetworkData<H, S, SS> = Message<SignableHash<H>, S, SS>;
+
+#[derive(Debug)]
+pub enum NetworkError {
+    SendFail,
+}
+
+#[async_trait::async_trait]
+pub trait DataNetwork<D>: Send + Sync {
```

Comment:
Welp, I really don't like that this is another instance of the same trait. Could we just use this instead of the one defined in `finality-aleph` everywhere?

Reply:
Alternatively at least fix `D = RmcNetworkData`? Although this will likely look uglier in code, since then this would have 3 type parameters...

Reply:
What about removing that trait at all from this crate and provide similar mechanism to that  of `rmc` from aleph_bft - API with some channels of data and consumer must do something with them. Then you need to move that network forwarding from aggregatorIO.

Reply:
I am ok(-ish) with copy-pasted impl in aggregator's crate. Suggestion to remove `<D>` is good, but has also weird cons: from type level perspective requiring it to be for `forall D` means it can't be specialized for specific `D=RmcNetworkData`. If we fix `D`, then aggregator itself, kind of, doesn't define a self-contained versioned protocol - finality-aleph might provide DataNetwork impl that do some `additional` things (change that doesn't depend on aggregator's api at all).

Reply:
So, actually channel based api doesn't fix that `problem` either. Maybe just change it's name from `DataNetwork` to something like `ProtocolSink` and everyone will be happy :)

Reply:
for now, I have renamed the trait to ProtocolSink but without fixing the D. in the aggregation code we do not need to know what is the D and trying to fix it to RmcNetworkData would result in introducing more type parameters that are actually not relevant to the code

Reply:
Ugh, this is still disgusting but at least swept under the rug by renaming. xd

I think that maybe the actual answer is to have a generic `DataNetwork` in AlephBFT and then use that everywhere, including changing the interface of `aleph-rmc`. I'll create a task...